### PR TITLE
Update hardware.rst

### PR DESCRIPTION
**Pull request by ErbB4**

### CODE REVIEW

File: doc/cpp/hardware.rst [hunk 1]
```diff
@@ -131,13 +131,13 @@ and handling exceptions on MPI exit.
...
     to initialize MPI with the minimum level thread support required by Arbor, that is
-    ``MPI_THREAD_SERIALIZED``. When it goes out of scope it will automatically call
+    ``MPI_THREAD_SERIALIZED``. When it goes out of scope, it will automatically call
     :cpp:any:`MPI_Finalize`.
 
     .. cpp:function:: with_mpi(int& argcp, char**& argvp, bool fatal_errors = true)
 
         The constructor takes the :cpp:any:`argc` and :cpp:any:`argv` arguments
-        passed to main of the calling application, and an additional flag
+        passed to the main of the calling application, and an additional flag
```
Comment (thorstenhater) [inline doc/cpp/hardware.rst:140]:
```suggestion
        passed to the ``main`` function of the calling application, and an additional flag
```

File: doc/cpp/hardware.rst [hunk 2]
```diff
@@ -329,13 +329,13 @@ whether it has a GPU, how many threads are in its thread pool, using helper func
 .. cpp:function:: unsigned num_ranks(const context&)
 
    Query the number of distributed ranks. If the context has an MPI
-   communicator, return is equivalent to :cpp:any:`MPI_Comm_size`.
+   communicator, the return is equivalent to :cpp:any:`MPI_Comm_size`.
```
Comment (thorstenhater) [inline doc/cpp/hardware.rst:332]:
```suggestion
   communicator, the return value is equivalent to :cpp:any:`MPI_Comm_size`.
```

File: doc/cpp/hardware.rst [hunk 3]
```diff
@@ -329,13 +329,13 @@ whether it has a GPU, how many threads are in its thread pool, using helper func
...
    If the communicator has no MPI, returns 1.
 
 .. cpp:function:: unsigned rank(const context&)
 
    Query the rank of the calling rank. If the context has an MPI
-   communicator, return is equivalent to :cpp:any:`MPI_Comm_rank`.
+   communicator, the return is equivalent to :cpp:any:`MPI_Comm_rank`.
```
Comment (thorstenhater) [inline doc/cpp/hardware.rst:338]:
```suggestion
   communicator, the return value is equivalent to :cpp:any:`MPI_Comm_rank`.
```

File: doc/cpp/hardware.rst [hunk 4]
```diff
@@ -271,14 +271,14 @@ The core Arbor library *libarbor* provides an API for:
 
         The identifier of the GPU to use.
         The gpu id corresponds to the ``int device`` parameter used by CUDA API calls
```
Comment (thorstenhater) [inline doc/cpp/hardware.rst:273]:
```suggestion
        The GPU id corresponds to the ``int device`` parameter used by CUDA API calls
```

File: doc/cpp/hardware.rst [hunk 5]
```diff
@@ -152,10 +152,10 @@ and handling exceptions on MPI exit.
         if the rank with the exception attempts to call :cpp:any:`MPI_Finalize` and
         other ranks are waiting in other MPI calls. If this happens inside a try-catch
         block, the deadlock stops the exception from being handled.
-        For this reason the destructor of :cpp:class:`with_mpi` only calls
+        For this reason, the destructor of :cpp:class:`with_mpi` only calls
         :cpp:any:`MPI_Finalize` if there are no uncaught exceptions.
-        This isn't perfect because the other MPI ranks still deadlock,
-        however it gives the exception handling code to report the error for debugging.
+        This isn't perfect because the other MPI ranks are still deadlock,
```
Comment (thorstenhater) [inline doc/cpp/hardware.rst:157]: Deadlock is used as a verb here ... not common outside of programmer jargon, but 'are' seems misplaced here. 'Can' could be an option.

**Review by thorstenhater — CHANGES_REQUESTED** [review]

File: doc/cpp/hardware.rst (hunk 2 above)
Comment (ErbB4) [reply doc/cpp/hardware.rst:332]: return value? returned value?

File: doc/cpp/hardware.rst (hunk 3 above)
Comment (ErbB4) [reply doc/cpp/hardware.rst:338]: same question here: return value? returned value?

File: doc/cpp/hardware.rst (hunk 5 above)
Comment (ErbB4) [reply doc/cpp/hardware.rst:157]: it is used many times in this file as a noun

File: doc/cpp/hardware.rst (hunk 2 above)
Comment (thorstenhater) [reply doc/cpp/hardware.rst:332]: Programming vocabulary: value returned by function.

File: doc/cpp/hardware.rst (hunk 5 above)
Comment (thorstenhater) [reply doc/cpp/hardware.rst:157]: Yes, but here it is a verb. A deadlock is the state in which multiple threads wait for each other (often circular) and none can make progress. To deadlock is to diverge into that state. Lingo, as I said ;)

**Review by thorstenhater — APPROVED** [review]

**Pull request merged**